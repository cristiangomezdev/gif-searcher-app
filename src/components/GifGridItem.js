import React from "react";
import PropTypes from 'prop-types';
export const GifGridItem = ({id,title,url})=>{
    return (
        <div className="card animate__animated animate__fadeIn animate__delay-0s"><img src={url} alt={title} key={id} ></img><p>{title}</p></div>)
}

GifGridItem.propTypes={
    id:PropTypes.string.isRequired,
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,
 }