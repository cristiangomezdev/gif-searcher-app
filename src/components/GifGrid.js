import React from "react";
import { GifGridItem } from "./GifGridItem";
//import { PropTypes } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
export const GifGrid = ({category})=>{
    const {data: images,loading} = useFetchGifs(category);

    return (<React.Fragment>
        <h3 className="animate__animated animate__fadeIn animate__delay-3s">{category}</h3>
        {loading && <p className="animate__animated animate__flash">Loading</p> }
        { <div className="card-grid">
        
            {images.map(image => <GifGridItem key={image.id} id={image.id} title={image.title} url={image.url} />)}
        </div> }
        </React.Fragment>)
}
//GifGrid.propTypes = { category: PropTypes.string.isRequired}