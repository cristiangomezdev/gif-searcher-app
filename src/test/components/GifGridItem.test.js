import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('counter app test', () => {

        const title='un titulo';
        const id='23123';
        const url = 'https://dynamic-media-cdn.tripadvisor.com/media/photo-o/15/a4/9b/77/legacy-hotel-at-img-academy.jpg?w=1200&h=-1&s=1';
        const wrapper = shallow(<GifGridItem id={id} title={title} url={url} /> )
    test('match con snapshot', () => {
 
            expect(wrapper).toMatchSnapshot();

    });
    test('debe de tener un parrafo con el title', () => {
       
            const p = wrapper.find('p');
            expect(p.text().trim()).toBe(title)
    });
    test('debe de tener la imagen igual al url y alt de los props', () => {
            const img = wrapper.find('img');

            expect(img.prop('src')).toBe(url)
            expect(img.prop('alt')).toBe(title)
    });
    test('debe tener el animate_fadein', () => {
            const div = wrapper.find('div');
                const className = div.prop('className');
            expect(className.includes('animate__fadeIn')).toBe(true);
    });
})