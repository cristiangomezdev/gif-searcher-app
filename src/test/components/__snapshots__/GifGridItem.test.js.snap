// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter app test match con snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-0s"
>
  <img
    alt="un titulo"
    key="23123"
    src="https://dynamic-media-cdn.tripadvisor.com/media/photo-o/15/a4/9b/77/legacy-hotel-at-img-academy.jpg?w=1200&h=-1&s=1"
  />
  <p>
    un titulo
  </p>
</div>
`;
